fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java

```diff
@@ -70,7 +69,9 @@ ScalarOperator simplifiedCaseWhenToIfFunction(CaseWhenOperator operator) {
         Type[] argTypes = args.stream().map(ScalarOperator::getType).toArray(Type[]::new);
         Function fn = Expr.getBuiltinFunction(FunctionSet.IF, argTypes, Function.CompareMode.IS_IDENTICAL);
 
-        Preconditions.checkState(fn != null);
```

Comment:
I think we should keep the check, which could help us find other bugs.

Reply:
The check will throw exception，but we can don't optimize the operator